### PR TITLE
u3d/install: support full pkg for Linux

### LINKED ISSUE #254: Error "Cancelled 'Build.Player.AndroidPlayer'" when Building APK
### Issue Checklist

- [X ] I'm using the latest u3d version
- [X ] I ran `u3d --help`
- [X ] I read the [README](https://github.com/DragonBox/u3d/blob/master/README.md)
- [X ] I made sure that a similar [issue](https://github.com/DragonBox/u3d/issues) doesn't exit

### Issue Description

[DESCRIBE YOUR ISSUE HERE]

Hi, thanks for let me post an issue...

i'm using : 
+Unity Personal Edition
+My Own installation of Unity 2017.3.0f1 and place it on **/opt/unity-editor-2017.3.0f1**
+Unity Build Scripts from [here](https://github.com/Anooky/unity-android-auto-build/blob/master/Assets/_scripts/Editor/BuildProject.cs), to produce an APK.

My Command when did this tasks : 

`u3d run -u 2017.3.0f1 -- -batchmode -quit -logFile build.log -nographics -batchmode -executeMethod" BuildProject.BuildAndroid" -androidSdkPath "/home/rain/android-sdk-linux" -buildOutput "/home/rain/myapps.apk" -projectPath "/home/rain/GameDev/myapps"  -buildPlatform android`

However, when i check its log, it come with below errors  :  

```
***** Cancelled 'Build.Player.AndroidPlayer' in 1 seconds (806 ms)
ERROR_CAUSED===>Error building player because build target was unsupported- Starting compile Library/ScriptAssemblies/Assem
bly-CSharp-Editor.dll
- Finished compile Library/ScriptAssemblies/Assembly-CSharp-Editor.dll**
```
any idea what caused this error ?
if because android module not installed, how to add it ?
or because it's Unity personal ?

Thanks in advance

### PR DESCRIPTION
**Pull request by lacostej**
Fixes #254

### CODE REVIEW

File: lib/u3d/installer.rb [hunk 1]
```diff
@@ -272,6 +277,35 @@ def install_xz(file, installation_path: nil)
...
+        U3dCore::CommandExecutor.execute(command: command)
+
+        path = pkg_install_relative_path("#{tmp_dir}/PackageInfo")
+
+        target_location = "#{installation_path}/#{path}"
+
+        # raise "Path for #{target_location} already exists" if path File.exist? target_location
+
+        command = "cd #{target_location.shellescape}; gzip -dc #{tmp_dir}/Payload | cpio -i * -"
+        command = "mkdir -p #{target_location.shellescape}; #{command}" # unless File.directory? installation_path
```
Comment (niezbop) [inline lib/u3d/installer.rb]: Use [`FileUtils.mkdir_p`](https://ruby-doc.org/stdlib-1.9.3/libdoc/fileutils/rdoc/FileUtils.html#method-c-mkdir_p)?

File: lib/u3d/installer.rb [hunk 2]
```diff
@@ -272,6 +277,35 @@ def install_xz(file, installation_path: nil)
...
+      raise 'Only able to install pkg on top of existing Unity installs' unless File.exist? installation_path
+      raise 'Missing 7z' if `which 7z`.empty?
+
+      Dir.mktmpdir do |tmp_dir|
+        UI.verbose "Working in tmp dir #{tmp_dir}"
+
+        command = "7z -aos -o#{tmp_dir.shellescape} e #{file.shellescape}"
+        U3dCore::CommandExecutor.execute(command: command)
+
+        path = pkg_install_relative_path("#{tmp_dir}/PackageInfo")
```
Comment (niezbop) [inline lib/u3d/installer.rb]: Would it be cleaner to wrap this call and the next one into a single method? ie

```ruby
path = pkg_install_path(installationPath, "#{tmp_dir}/PackageInfo")
```

File: lib/u3d/installer.rb (hunk 1 above)
Comment (lacostej) [reply lib/u3d/installer.rb]: yes. Will fix.

File: lib/u3d/installer.rb (hunk 2 above)
Comment (lacostej) [reply lib/u3d/installer.rb]: One is completely independent of having an installationPath in my head.

Comment (lacostej) [reply lib/u3d/installer.rb]: Changed my mind. Fixing.

**Comment by lacostej** [comment]
The implementation seems to work. If I diff a package installed using Unity's unattended installer and ours, this is the diff:
```
root@ubuntu-512mb-ams2-01:~# diff -ur /opt/unity-editor-2017.4.0f1/ unity-editor-2017.4.0f1/
Only in /opt/unity-editor-2017.4.0f1/Editor/Data/PlaybackEngines/iOSSupport/Trampoline/build: ._Unity-iPhone.build
Only in /opt/unity-editor-2017.4.0f1/Editor/Data/PlaybackEngines/iOSSupport/Trampoline: ._build
Only in /opt/unity-editor-2017.4.0f1/Editor/Data/Resources/Upm/__MACOSX: ._package.json
Only in /opt/unity-editor-2017.4.0f1/Editor/Data/Resources/Upm/__MACOSX: ._upm-linux
Only in /opt/unity-editor-2017.4.0f1/Editor/Data/Resources/Upm/__MACOSX: ._upm-macos
Only in /opt/unity-editor-2017.4.0f1/Editor/Data/Resources/Upm/__MACOSX: ._upm-win.exe
```

**Pull request merged**